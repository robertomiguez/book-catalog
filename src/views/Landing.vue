<template>
    <div class="leading">
        <div class="leading-text">{{ leadingText }}</div>
        <Button to='Book' text='Start here'/>
    </div>
</template>

<script>
import Button from '../components/Button'
export default {
  name: `Landing`,
  components: {
    Button
  },
  data: 
    function () {
      return {
        leadingText: `A website about book catalogs.`
      }
    }
}
</script>

<style scoped>
.leading {
  position:fixed;
  bottom:0;
  left:0;  
  width:100%;
  height: 100%;
  color: white;
  background: url('../assets/img/books.jpg') center #333 no-repeat;
  background-size: cover;
  color: white;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

@media (min-width: 800px) {
 .leading {
    display: flex;
    align-items: center;
  }
}

.leading-text {
  max-width: 250px;
  font-size: 1.3em;
  line-height: 1.5em;
  text-shadow: 1px 1px 3px #000;
  padding-bottom: 30px; 
}

@media (min-width: 800px) {
  .leading-text {
    max-width: 600px;
    font-size: 2em;
  }

}

</style>