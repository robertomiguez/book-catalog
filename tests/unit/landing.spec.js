import { shallowMount } from '@vue/test-utils'
import Landing from '@/views/Landing.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Landing, {
    propsData: {},
    mocks: {},
    stubs: {},
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe(`Landing`, () => {
  test(`testing component data`, () => {
    expect(wrapper.find(`.leading-text`).text())
      .toBe(`A website about book catalogs.`)
  })  

  test(`regression test`, () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })  

})