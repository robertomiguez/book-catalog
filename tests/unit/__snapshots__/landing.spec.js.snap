// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing regression test 1`] = `
<div
  class="leading"
>
  <div
    class="leading-text"
  >
    A website about book catalogs.
  </div>
   
  <button-stub
    text="Start here"
    to="Book"
  />
</div>
`;
